### PR TITLE
refine sparse momentum api and unitest

### PR DESCRIPTION
**Pull request by luotao1**

### CODE REVIEW

File: paddle/trainer/tests/sample_trainer_config_opt_b.conf [hunk 1]
```diff
@@ -12,32 +12,29 @@
...
-    learning_rate = 1e-03,
-    batch_size = 1000,
-    algorithm = 'sgd',
-    num_batches_per_send_parameter = 1,
-    num_batches_per_get_parameter = 1,
-    learning_method='momentum',
-)
-default_momentum(0.5)
+# default learning_method is MomentumOptimizer()
+settings(batch_size = 1000)
```
Comment (reyoung) [inline paddle/trainer/tests/sample_trainer_config_opt_b.conf]: There is no momentum set in configuration.

Comment (luotao1) [inline paddle/trainer/tests/sample_trainer_config_opt_b.conf]: set to 0.5 now.

File: paddle/trainer/tests/sample_trainer_config_opt_a.conf [hunk 2]
```diff
@@ -12,32 +12,29 @@
...
-    learning_rate = 1e-03,
-    batch_size = 1000,
-    algorithm = 'sgd',
-    num_batches_per_send_parameter = 1,
-    num_batches_per_get_parameter = 1,
-    learning_method='sparse_momentum',
-)
-default_momentum(0.5)
+settings(batch_size = 1000,
+         learning_method = MomentumOptimizer(Momentum=0.5, sparse=True))
```
Comment (reyoung) [inline paddle/trainer/tests/sample_trainer_config_opt_a.conf]: is `Momentum`?

https://github.com/baidu/Paddle/pull/126/files#diff-77565d531167fc12792a7aaedf18a243R106 code shows it is `momentum`.

Is this file pass unittest?

Comment (luotao1) [inline paddle/trainer/tests/sample_trainer_config_opt_a.conf]: In previous unittest, as `need_high_accuracy=true`, this unitest is always skip. Now, set `need_high_accuracy=false`, and find the `data_bin_part` has 9 inputs, while the config only has 2 inputs (data and label). Thus, replace `data_bin_part` with part of mnist data. And the unitest can be run sucessfully.
